src/main/java/bio/terra/service/dataset/DatasetService.java

```diff
@@ -316,9 +322,23 @@ public DatasetSummaryModel patch(
     if (!patchSucceeded) {
       throw new RuntimeException("Dataset was not updated");
     }
+    // TODO - what should we do if they're turning off secure monitoring?
+    if (patchRequest.isSecureMonitoringEnabled()) {
+      enableSecureMonitoring(id);
+    }
     return datasetDao.retrieveSummaryById(id).toModel();
   }
 
+  private void enableSecureMonitoring(UUID id) {
+    Dataset dataset = retrieve(id);
+    String projectId = dataset.getDatasetSummary().getDataProject();
+    try {
+      bufferService.refolderProjectToSecureFolder(projectId);
```

Comment:
Do we have a sense of how long this process takes?  The patch endpoint returns synchronously so could be nice to have a rough idea.

Reply:
Very good point, I'm not sure how long it would take. Hopefully moving this to a flight should handle this case, too.

Reply:
Agreed!